server/src/utils/callAutomationUtils.ts

```diff
@@ -0,0 +1,294 @@
+    },
+    transcriptionOptions: transcriptionOptions
+  };
+
+  const callbackUri = callAutomationConfig?.ServerHttpUrl;
+
+  /**
+   * Call automation needs to create the call.
+   */
+
+  const automationClient = getCallAutomationClient();
+  const roomsLocator: CallLocator = { kind: 'roomCallLocator', id: roomId };
+  const connectedCallResult = await automationClient.connectCall(roomsLocator, callbackUri ?? '', options);
+  console.log((await connectedCallResult.callConnection.getCallConnectionProperties()).serverCallId);
+  const serverCallId = (await connectedCallResult.callConnection.getCallConnectionProperties()).serverCallId;
+  if (!serverCallId) {
+    throw new Error('Server call id not found');
+  }
+  startTranscriptionForCall(serverCallId);
+};
+
+/**
+ * This is how we attach call automation to the call. We want to not start transcription here. but
+ * will start it later when the call connects.
+ * @param serverCallId - this can be fetched from the client call object with the handler in the '
+ * Call info object.
+ */
+export const connectRoomsCall = async (serverCallId: string): Promise<void> => {
+  const transcriptionOptions = {
+    transportUrl: callAutomationConfig?.ServerWebSocketUrl ?? '',
+    transportType: 'websocket',
+    locale: 'en-US',
+    startTranscription: false
+  };
+
+  const connectionId = Object.keys(CALLCONNECTION_ID_TO_CORRELATION_ID).find((key) =>
+    CALLCONNECTION_ID_TO_CORRELATION_ID[key].serverCallId.includes(serverCallId)
+  );
+
+  /**
+   * Check if the call automation client and connection for the call already exists. If it does, we don't need to create a new one.
+   */
+  if (connectionId) {
+    console.log('Call connection for call already exists:', connectionId);
+    return;
+  }
+
+  const options = {
+    callIntelligenceOptions: {
+      cognitiveServicesEndpoint: callAutomationConfig?.CognitionAPIEndpoint ?? ''
+    },
+    transcriptionOptions: transcriptionOptions
+  };
+  const res = await getCallAutomationClient().connectCall(
+    {
+      kind: 'serverCallLocator',
+      id: serverCallId
+    },
+    callAutomationConfig?.ServerHttpUrl ?? '',
+    options
+  );
+  console.log('Connect call result', res);
+  const callConnection = res.callConnection;
+  console.log('Call connection', callConnection);
+};
+
+export const startTranscriptionForCall = async (
+  callConnectionId: string,
+  options?: TranscriptionOptions
+): Promise<void> => {
+  console.log('Starting transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+
+  return await callConnection.getCallMedia().startTranscription(options);
+};
+
+export const stopTranscriptionForCall = async (callConnectionId: string): Promise<void> => {
+  console.log('Stopping transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
```

Comment:
Check if all the console logs are still necessary, if so, combine the double ones into a single console log